### PR TITLE
fix: support credentials and parameters with special characters

### PR DESCRIPTION
**Pull request by smoke**
Given Username and Password with special characters like
```
docker run --rm -it -v $(pwd)/../../cloud:/cloud -e REGISTRY_USERNAME="user with sp ' c \" rs" -e REGISTRY_PASSWORD="pass with sp ' c \" rs" -e REGISTRY_URL="http://example.com/charts" -e CHART_FOLDER="/example-chart" bsord/helm-push:with-this-fix
```

this will execute the following command with properly escaped Username and Password special characters
```
helm cm-push /tmp/tmp.NbFBDc/example-chart-1.2.3.tgz http://example.com/charts --username 'user with sp '\'' c " rs' --password 'pass with sp '\'' c " rs'
```

where before the same would execute the following command with badly escaped Username and Password
```
helm cm-push example-chart-1.2.3.tgz http://example.com/charts --username user with sp \' c '"' rs --password pass with sp \' c '"' rs
```
and throwing
`Error: This command needs 2 arguments: name of chart, name of chart repository (or repo URL)`

### CODE REVIEW

**Comment by smoke** [comment]
replaces https://github.com/bsord/helm-push/pull/23/files#diff-6f9d41d046756f0ddc2fcee0626bdb50100d12b88f293734eff742818e03efa2R85 
by building in `OUT_DIR=$(mktemp -d)`

**Comment by bsord** [comment]
@smoke can this be done without the eval statements?

**Comment by bsord** [comment]
@smoke it looks like this also removes the `helm lint .` command, was this intentional?

**Comment by smoke** [comment]
> @smoke can this be done without the eval statements?

Not that easy. I don't like it as well, but I've spent hours looking and trying different ways with no success.
For that to happen may be there is a way the parameters to be built using bash arrays.

**Comment by smoke** [comment]
> @smoke it looks like this also removes the `helm lint .` command, was this intentional?

Not intentional, I will revert that.

**Comment by smoke** [comment]
@bsord I have rebased the branch, reverted the `helm lint` also. Please review and merge it when you decide.

**Comment by almereyda** [comment]
> For that to happen may be there is a way the parameters to be built using bash arrays.

A similar use case often comes to my eyes when patching statements like ` echo ${POSTGRES_PASSWORD}| xargs -I% sh -c 'psql://${POSTGRES_USER}:%@localhost'` into pipelines. Good ordering and choice of single and double quotation marks next to using explicit character escapes where applicable, plus (anonymous) pipes and redirects can show good results.

File: entrypoint.sh [hunk 1]
```diff
@@ -79,8 +79,9 @@ if [ "$ADD_REPOSITORIES" != "" ]; then
...
 fi
 
-cd ${CHART_FOLDER}
+cd "${CHART_FOLDER}"
 helm lint .
-helm package . ${REGISTRY_APPVERSION} ${REGISTRY_VERSION} ${UPDATE_DEPENDENCIES}
-helm inspect chart *.tgz
-helm cm-push *.tgz ${REGISTRY_URL} ${REGISTRY_USERNAME} ${REGISTRY_PASSWORD} ${REGISTRY_ACCESS_TOKEN} ${FORCE}
+OUT_DIR=$(mktemp -d)
+eval $(echo helm package . -d "${OUT_DIR}" "${REGISTRY_APPVERSION}" "${REGISTRY_VERSION}" "${UPDATE_DEPENDENCIES}")
```
Comment (almereyda) [inline entrypoint.sh:85]: For me as an outsider, what is this line supposed to do?

It appears there is a reason why it is not possible to write this as

```suggestion
helm package . -d "${OUT_DIR}" "${REGISTRY_APPVERSION}" "${REGISTRY_VERSION}" "${UPDATE_DEPENDENCIES}"
```

instead? Or could

```suggestion
sh -c 'helm package . -d "${OUT_DIR}" ${REGISTRY_APPVERSION} ${REGISTRY_VERSION} ${UPDATE_DEPENDENCIES}'
```

be a way?

Comment (almereyda) [reply entrypoint.sh:85]: Or even shorter?

```suggestion
helm package . -d ${OUT_DIR} ${REGISTRY_APPVERSION} ${REGISTRY_VERSION} ${UPDATE_DEPENDENCIES}
```

What is the observed failure when approaching it this naively?

Comment (almereyda) [reply entrypoint.sh:85]: There's also

```suggestion
sh -c "helm package . -d '${OUT_DIR}' '${REGISTRY_APPVERSION}' '${REGISTRY_VERSION}' '${UPDATE_DEPENDENCIES}'"
```

which could probably help to work around the issue with escaping special characters.

**Pull request closed without merging**